### PR TITLE
SOLR-17186: Streaming query breaks if token contains backtick

### PR DESCRIPTION
**Pull request by rahulgoswami**
https://issues.apache.org/jira/browse/SOLR-17186

# Description

StreamExpressionParser today replaces ` with " indiscriminately which can break queries where the tokens themselves contain backtick. 

# Solution

Allow for backticks to be escaped in the streaming query and only replace non-escaped backticks with "

# Checklist

Please review the following and check all that apply:

- [+ ] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [ +] I have created a Jira issue and added the issue ID to my pull request title.
- [ +] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ +] I have developed this patch against the `main` branch.
- [ +] I have run `./gradlew check`.
- [ +] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/solrj-streaming/src/java/org/apache/solr/client/solrj/io/stream/expr/StreamExpressionParser.java [hunk 1]
```diff
@@ -132,15 +132,12 @@ private static StreamExpressionNamedParameter generateNamedParameterExpression(S
         }
       }
 
-      // If contains ` replace with "
+      // If contains ` replace with ". Preserves ` which are escaped.
```
Comment (epugh) [inline solr/solrj-streaming/src/java/org/apache/solr/client/solrj/io/stream/expr/StreamExpressionParser.java:135]: COuld we rework this comment?   NOt sure I understand the ` " .  Preserves` text...?

**Review by epugh — CHANGES_REQUESTED** [review]

File: solr/solrj-streaming/src/java/org/apache/solr/client/solrj/io/stream/expr/StreamExpressionParser.java (hunk 1 above)
Comment (rahulgoswami) [reply solr/solrj-streaming/src/java/org/apache/solr/client/solrj/io/stream/expr/StreamExpressionParser.java]: Done

File: solr/solrj-streaming/src/java/org/apache/solr/client/solrj/io/stream/expr/StreamExpressionParser.java [hunk 2]
```diff
@@ -132,17 +132,17 @@ private static StreamExpressionNamedParameter generateNamedParameterExpression(S
         }
       }
 
-      // If contains ` replace with "
+      // If contains ` replace with ", except when ` is escaped in the query as (\`)
```
Comment (epugh) [inline solr/solrj-streaming/src/java/org/apache/solr/client/solrj/io/stream/expr/StreamExpressionParser.java:135]: Nice!

**Comment by epugh** [comment]
Looking good...    I see additional fix...   I am running the tests...  Let me know when you feel it's "done done" ;-).   and if the tests pass I'll commit.

**Comment by rahulgoswami** [comment]
And thanks for taking time to review this Eric!

**Pull request merged**